src/nvim/eval.c

```diff
@@ -15140,8 +15140,11 @@ static void f_sockconnect(typval_T *argvars, typval_T *rettv, FunPtr fptr)
     return;
   }
 
+  snprintf((char *)IObuff, sizeof(IObuff), "%s:%" PRIdLINENR,
+           (sourcing_name ? sourcing_name : (char_u *)"?"),
+           (sourcing_name ? sourcing_lnum : 0));
   const char *error = NULL;
-  uint64_t id = channel_connect(tcp, address, 50, &error);
+  uint64_t id = channel_connect(tcp, address, 50, &error, (char *)IObuff);
```

Comment:
This is a cheesy stop-gap. Better approach will be to include this kind of info in a rich channels API.